src/app/forms/validators/custom-async-validators.ts

```diff
@@ -31,12 +31,20 @@ export class CustomAsyncValidators {
 
   static updateTestStationDetails(store: Store<State>): AsyncValidatorFn {
     return (control: AbstractControl): Observable<null> => {
-      store
-        .pipe(select(getTestStationFromProperty((control as CustomFormControl).meta.name as keyof TestStation, control.value)), take(1))
-        .subscribe(stations => {
-          stations && store.dispatch(updateTestStation({ payload: stations }));
-        });
-      return of(null);
+      return store.pipe(
+        select(getTestStationFromProperty((control as CustomFormControl).meta.name as keyof TestStation, control.value)),
+        take(1),
+        tap(stations => {
+          const testStationName = control.parent?.get('testStationName');
+          const testStationType = control.parent?.get('testStationType');
+          if (stations) {
+            testStationName && testStationName.setValue(stations.testStationName, { emitEvent: true, onlySelf: true });
+            testStationType && testStationType.setValue(stations.testStationType, { emitEvent: true, onlySelf: true });
+          }
+        }),
```

Comment:
The `tap` operator is used to perform a side-effect with the emitted value so that we can keep the map logic pure. Although we could throw a error and handle it in the `catchError`, returning `void` is not the correct use.

Reply:
The `return` is used to exit the scope based on the condition presented above. As tap does not alter the emitted value(s), it is safe to use.

But fair enough if you don't like it.